vmprof/reader.py

```diff
@@ -1,23 +1,51 @@
 from __future__ import print_function
 import re
 import struct
+import array
 import subprocess
 import sys
-
+from itertools import islice
+from six.moves import zip as izip
 
 PY3 = sys.version_info[0] >= 3
 
 WORD_SIZE = struct.calcsize('L')
 
 def read_word(fileobj):
+    """Read a single long from `fileobj`."""
     b = fileobj.read(WORD_SIZE)
     r = int(struct.unpack('l', b)[0])
     return r
 
+def read_words(fileobj, nwords):
+    """Read `nwords` longs from `fileobj`."""
+    try:
+        # Do we have real file object?
+        fileobj.fileno()
```

Comment:
Whoops, this is an old version of the code, no idea why I committed this. But your suggestion is better anyways, thanks for the pointer!